AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/MFA/FetchMFAPreferenceTaskTests.swift

```diff
@@ -0,0 +1,356 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+import XCTest
+import Amplify
+@testable import AWSCognitoAuthPlugin
+import AWSCognitoIdentityProvider
+
+// swiftlint:disable type_body_length
+// swiftlint:disable file_length
+class FetchMFAPreferenceTaskTests: BasePluginTest {
+
+    /// Test a successful fetchMFAPreference call
+    ///
+    /// - Given: an auth plugin with mocked service. Mocked service calls should mock a successul response
+    /// - When:
+    ///    - I invoke fetchMFAPreference
+    /// - Then:
+    ///    - I should get a successful result
+    ///
+    func testSuccessfulPreferenceFetchWithTOTPPreferred() async {
```

Comment:
Should we add a test where service returns a new MFA type or invalid strings?

Reply:
Sure.. Will add it in a follow up PR.